### PR TITLE
test: add crossbow test

### PR DESCRIPTION
**Pull request by JoenvandeVorle**
## Description

Adds an aplib test for the crossbow weapon

## Testability

Run the tests. This time, they also work in the pipeline!

## Checklist
- [x] Set the proper pull request name
- [x] Merged main into your branch
- [x] Added a scene to the game for your changes

### CODE REVIEW

File: aplib.net-demo/Assets/Testing/AplibTests/RangedWeaponTest.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+namespace Testing.AplibTests
+{
+    public class RangeedWeaponTestBeliefSet : BeliefSet
+    {
+        /// <summary>
+        /// The player rotation object in the scene.
+        /// </summary>
+        public Belief<GameObject, GameObject> PlayerRotation = new(GameObject.Find("PlayerRotation"), x => x);
+
+        public Belief<GameObject, GameObject> Player = new(GameObject.Find("Player"), x => x);
```
Comment (Tboefijn) [inline aplib.net-demo/Assets/Testing/AplibTests/RangedWeaponTest.cs:24]: I am missing a summary here.

File: aplib.net-demo/Assets/Testing/AplibTests/RangedWeaponTest.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        /// </summary>
+        public Belief<GameObject, Vector3> EnemyPosition = new(GameObject.Find("Target Dummy Body"),
+            x => x != null ? x.transform.position : Vector3.zero);
+
+        /// <summary>
+        /// Bool that returns if enemy is dead or alive.
+        /// </summary>
+        public Belief<GameObject, bool> IsEnemyDead = new(GameObject.Find("Target Dummy Body"), x => x == null);
+
+        public Belief<GameObject, bool> IsEnemyInFront = new(
```
Comment (Tboefijn) [inline aplib.net-demo/Assets/Testing/AplibTests/RangedWeaponTest.cs:47]: I am missing a summary here.

File: aplib.net-demo/Assets/Testing/AplibTests/RangedWeaponTest.cs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            reference: GameObject.Find("PlayerRotation"),
+            getObservationFromReference: x =>
+            {
+                if (!Physics.Raycast(x.transform.position, x.transform.forward, out RaycastHit hit, 100))
+                {
+                    Debug.Log("No hit");
+                    return false;
+                }
```
Comment (Tboefijn) [inline aplib.net-demo/Assets/Testing/AplibTests/RangedWeaponTest.cs:55]: Deze Log is overbodig.
```suggestion
                if (!Physics.Raycast(x.transform.position, x.transform.forward, out RaycastHit hit, 100)) return false;
```

File: aplib.net-demo/Assets/Testing/AplibTests/RangedWeaponTest.cs [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+                return hit.collider.gameObject.name == "Target Dummy Body";
+            }
+        );
+    }
+
+    public class RangedWeaponTest
```
Comment (Tboefijn) [inline aplib.net-demo/Assets/Testing/AplibTests/RangedWeaponTest.cs:62]: Ik mis 1 of meerdere summaries die uitleggen wat deze test doet.

File: aplib.net-demo/Assets/Testing/AplibTests/RangedWeaponTest.cs [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+            // Assert that the player has reached the target position
+            Assert.IsTrue(condition: agent.Status == CompletionStatus.Success);
+            yield break;
+
+            bool EnemyKilledPredicate(RangeedWeaponTestBeliefSet beliefSet)
+            {
+                // The player has killed the enemy
+                bool enemyDead = beliefSet.IsEnemyDead;
+                return enemyDead;
+            }
```
Comment (Tboefijn) [inline aplib.net-demo/Assets/Testing/AplibTests/RangedWeaponTest.cs:129]:
```suggestion
            bool EnemyKilledPredicate(RangeedWeaponTestBeliefSet beliefSet) => beliefSet.IsEnemyDead;
```

File: aplib.net-demo/Assets/Testing/AplibTests/RangedWeaponTest.cs [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+            bool IsEnemyInFrontPredicate(RangeedWeaponTestBeliefSet beliefSet)
+            {
+                // The enemy is in front of the player
+                bool enemyInFront = beliefSet.IsEnemyInFront;
+                return enemyInFront;
+            }
```
Comment (Tboefijn) [inline aplib.net-demo/Assets/Testing/AplibTests/RangedWeaponTest.cs:136]:
```suggestion
            bool IsEnemyInFrontPredicate(RangeedWeaponTestBeliefSet beliefSet) => beliefSet.IsEnemyInFront;
```

**Review by Tboefijn — CHANGES_REQUESTED** [review]
Looks good! Left some small comments.

**Review by Tboefijn — APPROVED** [review]
LGTM!

**Review by SSelinn — APPROVED** [review]
Passed the test

**Pull request merged**